analyzers/src/SonarAnalyzer.CSharp/Rules/ConditionalsShouldStartOnNewLine.cs

```diff
@@ -46,10 +46,7 @@ protected override void Initialize(SonarAnalysisContext context)
                 if (TryGetPreviousTokenInSameLine(ifKeyword, out var previousTokenInSameLine) &&
                     previousTokenInSameLine.IsKind(SyntaxKind.CloseBraceToken))
                 {
-                    c.ReportIssue(Diagnostic.Create(rule,
-                            ifKeyword.GetLocation(),
-                            additionalLocations: new[] { previousTokenInSameLine.GetLocation() }
-                        ));
+                    c.ReportIssue(DiagnosticFactory.Create(rule, c.Compilation, ifKeyword.GetLocation(), additionalLocations: new[] { previousTokenInSameLine.GetLocation() }));
```

Comment:
`previousTokenInSameLine` suggests, that we can not end up in another compilation.

Reply:
and at the same time, I think it's better to not have exceptions and use the new helper method everywhere. wdyt?

Reply:
I would even force the use of the factory in all cases. Even those that do not use additionalLocation